llarp/linux/service_manager.cpp

```diff
@@ -0,0 +1,61 @@
+#include <llarp/util/service_manager.hpp>
+
+#if defined(WITH_SYSTEMD)
+#include <systemd/sd-daemon.h>
+#include <cassert>
+#endif
+
+namespace llarp::sys
+{
+#if defined(WITH_SYSTEMD)
+  class SD_Manager : public I_SystemLayerManager
+  {
+    llarp::sys::ServiceState m_State{ServiceState::Initial};
+
+   public:
+    /// change our state and report it to the system layer
+    void
+    we_changed_our_state(ServiceState st) override
+    {
+      assert(m_State != st);
+      m_State = st;
+      report_our_state();
+    }
+
+    /// report our current state to the system layer
+    void
+    report_our_state() override
+    {
+      if (m_State == ServiceState::Starting)
+      {
+        // TODO: maybe request more time?
+        ::sd_notify(0, "READY=0");
```

Comment:
This isn't right; from sd_notify docs:

> Since there is little value in signaling non-readiness, the only value services should send is "READY=1" (i.e. "READY=0" is not defined).